FormalConjectures/ErdosProblems/312.lean

```diff
@@ -0,0 +1,97 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Erdős Problem 312 — Fintype Formulation
+
+*Reference:* https://www.erdosproblems.com/312
+
+We state a version of the problem using functions `Fin n → ℕ` to represent a finite
+indexed family of positive integers and `Finset (Fin n)` for chosen subsets,
+making multiplicities explicit by indexing with `Fin n`.
+-/
+
+open scoped BigOperators
+
+namespace Erdos312
+
+noncomputable section
+
+variable {n : ℕ}
+
+/-- A short name for a finite family of (nonnegative) integers indexed by `Fin n`. -/
+abbrev Family (n : ℕ) := Fin n → ℕ
```

Comment:
Here it's more conventional to just inline `Fin n → ℕ` everywhere rather than using `Family n`.